### ISSUE TITLE
always in offline mode in v6.

### ISSUE THREAD

**Issue by gilbert-hsu** [ansible/ansible-lint#3565]
##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```console (paste below)

```

##### Desired Behavior

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`

```paste below

```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by ssbarnea** [comment]
I am inclined to say that this is only a misleading message as they linter would not really run in offline more. Very easy to test, remove one dependency and run again, likely I will not fail due not finding it. Still, we need to fix that too.

That is related to `get_app()` which sadly instantiates twice, 2nd one being for syntax check. I will try to come up with a fix till the end of this week, it is bit delicate to fix it correctly.

**Comment by gilbert-hsu** [comment]
> I am inclined to say that this is only a misleading message as they linter would not really run in offline more. Very easy to test, remove one dependency and run again, likely I will not fail due not finding it. Still, we need to fix that too.
> 

Thanks, 
You are right ! 
When testing with total cleaned environment, roles / collections has been downloaded

**Comment by max06** [comment]
@ssbarnea Are you sure this is fixed? I've just tested the current main branch and I still get to see the offline-message...

```text
 $  ansible-lint roles/base/tasks/main.yml -v                                                                                                                                                                 
INFO     Identified /workspaces/someproject/ansible as project root due config file /workspaces/someproject/ansible/.ansible-lint.
INFO     Running ansible-galaxy role install -vr requirements.yml --roles-path /home/vscode/.cache/ansible-compat/0cb87f/roles
INFO     Running ansible-galaxy collection install -v -r requirements.yml
INFO     Set ANSIBLE_LIBRARY=/home/vscode/.cache/ansible-compat/0cb87f/modules:/home/vscode/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/vscode/.cache/ansible-compat/0cb87f/collections:/home/vscode/.cache/ansible-compat/0cb87f/collections:/home/vscode/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/vscode/.cache/ansible-compat/0cb87f/roles:roles:/workspaces/someproject/ansible/roles
WARNING  Skipped installing old role dependencies due to running in offline mode.
WARNING  Skipped installing collection dependencies due to running in offline mode.
INFO     Set ANSIBLE_LIBRARY=/home/vscode/.cache/ansible-compat/0cb87f/modules:/home/vscode/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/vscode/.cache/ansible-compat/0cb87f/collections:/home/vscode/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/vscode/.cache/ansible-compat/0cb87f/roles:roles:/workspaces/someproject/ansible/roles

Passed: 0 failure(s), 0 warning(s) on 4 files. Profile 'shared' was required, but only 'production' profile passed.
You are using a pre-release version of ansible-lint.

ansible-lint 6.17.3.dev6 using ansible-core:2.15.1 ansible-compat:4.1.2 ruamel-yaml:0.17.32 ruamel-yaml-clib:0.2.7
```
